src/session/model/Session.ts

```diff
@@ -0,0 +1,127 @@
+import { Either, Left, Right } from "purify-ts";
+import { AccessTokenMissingError, ExpiresMissingError, SessionExpiredError, SignInInfoMissingError } from "../../error/ErrorFunctions";
+import { Failure } from "../../error/FailureType";
+import { SessionKeys } from "../SessionKeys";
+import { AccessToken } from "./AccessToken";
+import { IMap } from "./ISignInInfo";
+import { Cookie } from "./Cookie";
+import { SessionHandlerConfig } from "../../SessionHandlerConfig";
+
+export class Session {
+
+    public data: IMap<any> = {};
+
+    public constructor(data?: any) {
+
+        if (data) {
+
+            Session.marshall(this, data);
+        }
+
+    }
+
+
+    public unmarshall = () => {
+
+        const obj: any = {};
+        const thisObj: any = this.data;
+
+        const keys = Object.keys(thisObj).sort();
+        for (const i in keys) {
+            if (thisObj.hasOwnProperty(keys[i])) {
+                obj[keys[i]] = thisObj[keys[i]];
+
+            }
+        }
+
+        return obj;
+    };
+
+    public verify = (): Either<Failure, VerifiedSession> => {
+        return VerifiedSession.verifySession(this);
+    };
+
+    public static marshall = (session: Session, data: any): void => {
+        const keys = Object.keys(data).sort();
+
+        for (const i in keys) {
+            if (data.hasOwnProperty(keys[i])) {
+                session.data[keys[i]] = data[keys[i]];
+            }
+        }
+    };
+
+}
+
+export class VerifiedSession extends Session {
+
+    private constructor(session: Session) {
+        super();
+        this.data = session.data;
+    }
+
+    public asCookie = (): Cookie => {
+        return Cookie.sessionCookie(this);
+    };
+
+    public static createNewVerifiedSession(
```

Comment:
At this point I see no use case for it. It also looks wrong to issue 'verified' session while access tokens are dummy. Isn't accounts component the only one entitled to session creation?